nav2_util/src/lifecycle_service_client.cpp

```diff
@@ -67,33 +67,26 @@ LifecycleServiceClient::LifecycleServiceClient(
 
 bool LifecycleServiceClient::change_state(
   const uint8_t transition,
-  const seconds timeout)
+  const milliseconds transition_timeout,
+  const milliseconds wait_for_service_timeout)
 {
-  if (!change_state_.wait_for_service(timeout)) {
+  if (!change_state_.wait_for_service(wait_for_service_timeout)) {
     throw std::runtime_error("change_state service is not available!");
   }
 
   auto request = std::make_shared<lifecycle_msgs::srv::ChangeState::Request>();
   request->transition.id = transition;
-  auto response = change_state_.invoke(request, timeout);
-  return response.get();
-}
-
-bool LifecycleServiceClient::change_state(
-  std::uint8_t transition)
-{
-  if (!change_state_.wait_for_service(5s)) {
-    throw std::runtime_error("change_state service is not available!");
+  if (transition_timeout > milliseconds(0)) {
```

Comment:
```suggestion
  if (transition_timeout > 0ms) {
```

This should work and look nicer (pendantic)

Reply:
Looks like your fork changed so I can't merge my own suggestions as a maintainer - please merge this in and I can merge!